datastore/src/main/java/com/ds/datastore/BookController.java

```diff
@@ -20,35 +24,45 @@
 
     private final BookModelAssembler assembler;
 
+
+
     public BookController(BookRepository repository, BookModelAssembler assembler) {
         this.repository = repository;
         this.assembler = assembler;
     }
 
-    @PostMapping("/books")
-    protected ResponseEntity<EntityModel<Book>> newBook(@RequestBody Book book){
+    @PostMapping("/bookstores/{store_id}/books")
+    protected ResponseEntity<EntityModel<Book>> newBook(@RequestBody Book book, @PathVariable Long store_id){
+        book.setStore_id(store_id);
         EntityModel<Book> entityModel = assembler.toModel(repository.save(book));
+
         return ResponseEntity
                 .created(entityModel.getRequiredLink(IanaLinkRelations.SELF).toUri())
                 .body(entityModel);
     }
     
-    @GetMapping("/books/{id}")
-    protected EntityModel<Book> one(@PathVariable Long id) {
+    @GetMapping("/bookstores/{store_id}/books/{id}")
+    protected EntityModel<Book> one(@PathVariable Long id, @PathVariable Long store_id) {
         Book book = repository.findById(id).orElseThrow(() -> new BookNotFoundException(id));
+
         return assembler.toModel(book);
+
     }
 
-    @GetMapping("/books")
-    protected CollectionModel<EntityModel<Book>> all(){
-        List<EntityModel<Book>> books = repository.findAll().stream()
+    @GetMapping("/bookstores/{store_id}/books")
+    protected CollectionModel<EntityModel<Book>> all(@PathVariable long store_id){
+        List<Book> books = repository.findAll();
+        books.removeIf(book -> book.getStore_id() == null || book.getStore_id() != store_id);
```

Comment:
This seems like it would work... but I have to think there is a better way using the `repository`. Imagine if there are a million books - you are getting all million books and then filtering in memory. That seems very inefficient and could also cause your app to crash with an out of memory exception. I would look into some possibilities with the `repository`